map/search_api.cpp

```diff
@@ -0,0 +1,295 @@
+#include "map/search_api.hpp"
+
+#include "search/everywhere_search_params.hpp"
+#include "search/geometry_utils.hpp"
+#include "search/hotels_filter.hpp"
+#include "search/viewport_search_params.hpp"
+
+#include "storage/downloader_search_params.hpp"
+
+#include "platform/preferred_languages.hpp"
+
+#include "geometry/mercator.hpp"
+
+#include "base/string_utils.hpp"
+
+#include <vector>
+
+using namespace search;
+using namespace std;
+
+namespace
+{
+double const kDistEqualQueryMeters = 100.0;
+
+// Cancels search query by |handle|.
+void CancelQuery(weak_ptr<ProcessorHandle> & handle)
+{
+  if (auto queryHandle = handle.lock())
+    queryHandle->Cancel();
+  handle.reset();
+}
+
+bool IsCianMode(string query)
+{
+  strings::Trim(query);
+  strings::AsciiToLower(query);
+  return query == "cian";
+}
+}  // namespace
+
+SearchAPI::SearchAPI(Index & index, storage::Storage const & storage,
+                     storage::CountryInfoGetter const & infoGetter, Delegate & delegate)
+  : m_index(index)
+  , m_storage(storage)
+  , m_infoGetter(infoGetter)
+  , m_delegate(delegate)
+  , m_engine(m_index, GetDefaultCategories(), m_infoGetter,
+             Engine::Params(languages::GetCurrentOrig() /* locale */, 1 /* params.m_numThreads */))
+{
+}
+
+void SearchAPI::OnViewportChanged(m2::RectD const & viewport)
+{
+  m_viewport = viewport;
+
+  auto const forceSearchInViewport = !m_isViewportInitialized;
+  if (!m_isViewportInitialized)
```

Comment:
if (forceSearchInViewport) ?

Reply:
No. Please, take a look at the body of the conditional - it should be executed when viewport is not I.

Reply:
Is not initialized.